packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -442,6 +442,10 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
 
   private _renderNormalItem(item: IContextualMenuItem, classNames: IMenuItemClassNames, index: number, focusableElementIndex: number, totalItemCount: number, hasCheckmarks: boolean, hasIcons: boolean): React.ReactNode {
     if (item.onRender) {
+      item.positionInSet = focusableElementIndex;
```

Comment:
If we do proceed with this we will need to allow users to provide their own values for this. So you should do an expand like `const itemWithProps = {... {positionInSet: focusableElementIndex, setSize: totalItemCount, hasCheckMarks: hasCheckMarks, hasIcon: hasIcons}, ...item}` that way their passed in item values will override the ones you provide.

Reply:
I agree that we should just spread the props, assign the value and allow for it to be overridden. You wouldn't need to put the other props inside of their own object though, it could look something like:
`item.onRender( {posInSet: focusableElementIndex, setSize: totalItemCount, ...item}, this.dismiss)`

Reply:
I may not be understanding correctly, but it seems like we're mutating the item, which is a terrible idea. We shouldn't pollute the consumer input like this. having some additional props in the onRender stuff seems reasonable though.

Reply:
Perfect, I switched to use the spread operator.